extensions/amp-a4a/0.1/signature-verifier.js

```diff
@@ -0,0 +1,357 @@
+   * for each key to the `keys` object.
+   *
+   * @param {!Object<string, ?Promise<?webCrypto.CryptoKey>>} keys the object to
+   *     add each key promise to. This is mutated while the returned promise is
+   *     pending.
+   * @param {string} signingServiceName
+   * @param {?string} keypairId the keypair ID to include in the query string
+   *     for cachebusting purposes, or `null` if no cachebusting is needed
+   * @return {!Promise<boolean>} resolves after the mutation of `keys` is
+   *     complete, to `true` if the keyset was downloaded and parsed
+   *     successfully (even if some keys were malformed), or `false` if a
+   *     keyset-level failure occurred
+   * @private
+   */
+  fetchAndAddKeys_(keys, signingServiceName, keypairId) {
+    let url = signingServerURLs[signingServiceName];
+    if (keypairId != null) {
+      url += '?' + encodeURIComponent(keypairId);
+    }
+    return xhrFor(this.win_)
+        .fetch(
+            url,
+            {mode: 'cors', method: 'GET', ampCors: false, credentials: 'omit'})
+        .then(
+            response => {
+              if (response.status == 200) {
+                // The signing service spec requires this, but checking it at
+                // runtime in production isn't worth it.
+                dev().assert(
+                    response.headers.get('Content-Type') ==
+                    'application/jwk-set+json');
+                return response.json().then(
+                    jwkSet => {
+                      // This is supposed to be a JSON Web Key Set, as defined
+                      // in Section 5 of RFC 7517. However, the signing service
+                      // could misbehave and send an arbitrary JSON value, so we
+                      // have to type-check at runtime.
+                      if (jwkSet && Array.isArray(jwkSet['keys'])) {
+                        jwkSet['keys'].forEach(jwk => {
+                          if (jwk && typeof jwk['kid'] == 'string') {
+                            if (keys[jwk['kid']] === undefined) {
+                              // We haven't seen this keypair ID before.
+                              keys[jwk['kid']] =
+                                  cryptoFor(this.win_).importPkcsKey(jwk).catch(
+                                      err => {
+                                        // Web Cryptography rejected the key
+                                        // import attempt. Either the signing
+                                        // service sent a malformed key or the
+                                        // browser is doing something weird.
+                                        const jwkData = JSON.stringify(jwk);
+                                        const message = err && err.message;
+                                        signingServiceError(
+                                            signingServiceName,
+                                            `Failed to import key (${
```

Comment:
I agree, but clang-format doesn't. I'm hesitant to contradict it because that might result in the line getting reformatted later.